lib/pure/options.nim

```diff
@@ -79,123 +72,302 @@ type
+        a: int
+        b: string
+    var c: Foo
+    assert c.isNil
+    var d = option(c)
+    assert d.isNone
+
+  result.val = val
+  when T isnot SomePointer:
+    result.has = true
+
 proc some*[T](val: T): Option[T] =
-  ## Returns a ``Option`` that has this value.
+  ## Returns an ``Option`` that has the value `val`.
+  ##
+  ## See also:
+  ## * `option <#option,T>`_
+  ## * `none <#none,typedesc>`_
+  ## * `isSome <#isSome,Option[T]>`_
+  runnableExamples:
+    var
+      a = some("abc")
+      b = some(42)
+    assert $type(a) == "Option[system.string]"
+    assert b.isSome
+    assert a.get == "abc"
+    assert $b == "Some(42)"
+
   when T is SomePointer:
     assert val != nil
     result.val = val
   else:
     result.has = true
     result.val = val
 
-proc option*[T](val: T): Option[T] =
-  ## Can be used to convert a pointer type to an option type. It
-  ## converts ``nil`` to the none-option.
-  result.val = val
-  when T isnot SomePointer:
-    result.has = true
-
 proc none*(T: typedesc): Option[T] =
   ## Returns an ``Option`` for this type that has no value.
+  ##
+  ## See also:
+  ## * `option <#option,T>`_
+  ## * `some <#some,T>`_
+  ## * `isNone <#isNone,Option[T]>`_
+  runnableExamples:
+    var a = none(int)
+    assert a.isNone
+    assert $type(a) == "Option[system.int]"
+
   # the default is the none type
   discard
 
 proc none*[T]: Option[T] =
-  ## Alias for ``none(T)``.
+  ## Alias for `none(T) proc <#none,typedesc>`_.
   none(T)
 
 proc isSome*[T](self: Option[T]): bool {.inline.} =
+  ## Checks if an ``Option`` contains a value.
+  runnableExamples:
+    var
+      a = some(42)
+      b = none(string)
+    assert a.isSome
+    assert not b.isSome
+
   when T is SomePointer:
     self.val != nil
   else:
     self.has
 
 proc isNone*[T](self: Option[T]): bool {.inline.} =
+  ## Checks if an ``Option`` is empty.
+  runnableExamples:
+    var
+      a = some(42)
+      b = none(string)
+    assert not a.isNone
+    assert b.isNone
   when T is SomePointer:
     self.val == nil
   else:
     not self.has
 
-proc unsafeGet*[T](self: Option[T]): T =
-  ## Returns the value of a ``some``. Behavior is undefined for ``none``.
-  assert self.isSome
-  self.val
-
 proc get*[T](self: Option[T]): T =
-  ## Returns contents of the Option. If it is none, then an exception is
+  ## Returns contents of an ``Option``. If it is ``None``, then an exception is
   ## thrown.
+  ##
+  ## See also:
+  ## * `get proc <#get,Option[T],T>`_ with the default return value
+  runnableExamples:
+    let
+      a = some(42)
+      b = none(string)
+    assert a.get == 42
+    doAssertRaises(UnpackError):
+      echo b.get
+
   if self.isNone:
     raise newException(UnpackError, "Can't obtain a value from a `none`")
   self.val
 
 proc get*[T](self: Option[T], otherwise: T): T =
-  ## Returns the contents of this option or `otherwise` if the option is none.
+  ## Returns the contents of the ``Option`` or an `otherwise` value if
```

Comment:
While single-backtick code markup is supported now double vs single should be consistent in Nim's docs IMO.

Single has the benefit of being easier to turn into an RST style link, aesthetics/ergonomics aside.

Reply:
The existing docs was a mixture of single- and double-backticks, so I decided to keep it that way.

But yeah, I'll change everything to single-backticks to be consistent.